code/modules/mod/mod_activation.dm

```diff
@@ -86,9 +86,21 @@
 	if(part_datum.can_overslot)
 		var/obj/item/overslot = wearer.get_item_by_slot(part.slot_flags)
 		if(overslot)
+			if(istype(overslot, /obj/item/mod))
+				balloon_alert(user, "can't overslot MOD suit parts!")
+				playsound(src, 'sound/machines/scanner/scanbuzz.ogg', 25, TRUE, SILENCED_SOUND_EXTRARANGE)
+				return FALSE
+			if(HAS_TRAIT(overslot, TRAIT_NODROP))
+				balloon_alert(user, "can't remove [overslot]!")
+				playsound(src, 'sound/machines/scanner/scanbuzz.ogg', 25, TRUE, SILENCED_SOUND_EXTRARANGE)
+				return FALSE
+			if(!overslot.forceMove(src))
+				balloon_alert(user, "failed to store old part!")
+				playsound(src, 'sound/machines/scanner/scanbuzz.ogg', 25, TRUE, SILENCED_SOUND_EXTRARANGE)
+				return FALSE
 			part_datum.overslotting = overslot
 			wearer.transferItemToLoc(overslot, part, force = TRUE)
-			RegisterSignal(part, COMSIG_ATOM_EXITED, PROC_REF(on_overslot_exit))
+			RegisterSignal(part, COMSIG_ATOM_EXITED, PROC_REF(on_overslot_exit), override = TRUE)
```

Comment:
If you need to use override, you most likely have a flaw in your op order somewhere, exceptions are rare.